README.md
# elk_beats Cookbook

TODO: Enter the cookbook description here.

e.g.
This cookbook makes your favorite breakfast sandwich.

## Requirements

TODO: List your cookbook requirements. Be sure to include any requirements this cookbook has on platforms, libraries, other cookbooks, packages, operating systems, etc.

e.g.
### Platforms

- SandwichOS

### Chef

- Chef 12.0 or later

### Cookbooks

- `toaster` - ibm_beats needs toaster to brown your bagel.

## Attributes

TODO: List your cookbook attributes here.

e.g.
### ibm_beats::default

<table>
  <tr>
    <th>Key</th>
    <th>Type</th>
    <th>Description</th>
    <th>Default</th>
  </tr>
  <tr>
    <td><tt>['ibm_beats']['bacon']</tt></td>
    <td>Boolean</td>
    <td>whether to include bacon</td>
    <td><tt>true</tt></td>
  </tr>
</table>

## Usage

### ibm_beats::default

TODO: Write usage instructions for each cookbook.

e.g.
Just include `ibm_beats` in your node's `run_list`:

```json
{
  "name":"my_node",
  "run_list": [
    "recipe[ibm_beats]"
  ]
}
```

## Contributing

TODO: (optional) If this is a public cookbook, detail the process for contributing. If this is a private cookbook, remove this section.

e.g.
1. Fork the repository on Github
2. Create a named feature branch (like `add_component_x`)
3. Write your change
4. Write tests for your change (if applicable)
5. Run the tests, ensuring they all pass
6. Submit a Pull Request using Github

## License and Authors

Authors: TODO: List authors



CHANGELOG.md
# beats CHANGELOG

This file is used to list changes made in each version of the openssh cookbook.

## 0.1.0
- [your_name] - Initial release of openssh

- - -
Check the [Markdown Syntax Guide](http://daringfireball.net/projects/markdown/syntax) for help with Markdown.

The [Github Flavored Markdown page](http://github.github.com/github-flavored-markdown/) describes the differences between markdown on github and standard markdown.


recipes/install_filebeat.rb
########################################################################################################################
#                                                                                                                      #
#                                  OpenSSH Cookbook           											               #
#                                                                                                                      #
#   Language            : Chef/Ruby                                                                                    #
#   Date                : 11/28/2017                                                                                   #
#   Date Last Update    : 11/28/2017                                                                                   #
#   Version             : 1.0                                                                                          #
#   Author              : Arnaud Thalamot                                  											   #
#                                                                                                                      #
########################################################################################################################

#Windows Beats
if platform_family?('windows')
end

#Linux Beats
if platform_family?('rhel')

  remote_file '/tmp/filebeat-7.4.0-x86_64.rpm' do
        source 'https://client.com/ibm/chef-client/filebeat-7.4.0-x86_64.rpm'
        owner 'root'
        group 'root'
        mode '0755'
        action :create_if_missing
    end

    # Install metricbeats rpm
	execute 'install filebeat' do
        command 'rpm -ivh /tmp/filebeat-7.4.0-x86_64.rpm'
        action :run
        only_if { shell_out("rpm -qa filebeat").stdout == ''}
    end

    service 'filebeat' do
      supports :status => true
      action :start
    end
end

recipes/install_metricbeat.rb
########################################################################################################################
#                                                                                                                      #
#                                  ELK Beats Cookbook          											               #
#                                                                                                                      #
#   Language            : Chef/Ruby                                                                                    #
#   Date                : 11/28/2017                                                                                   #
#   Date Last Update    : 11/28/2017                                                                                   #
#   Version             : 1.0                                                                                          #
#   Author              : Arnaud Thalamot                                                                              #
#                                                                                                                      #
########################################################################################################################

#Windows Beats
if platform_family?('windows')

  remote_file 'C:/metricbeat-7.4.0-windows-x86_64.zip' do
        source 'https://client.com/ibm/chef-client/metricbeat-7.4.0-windows-x86_64'
        owner 'root'
        group 'root'
        mode '0755'
        action :create_if_missing
    end

  directory 'C:/metricbeat' do
    action :create_if_missing
  end

    ruby_block 'unzip-install-file' do
          block do
            command = powershell_out "Add-Type -assembly \"system.io.compression.filesystem\"; [io.compression.zipfile]::ExtractToDirectory('C:/metricbeat-7.4.0-windows-x86_64.zip', 'C:/metricbeat')"
            Chef::Log.debug command.to_s
            action :create
          end
        end

end

#Linux Beats
if platform_family?('rhel')

  remote_file '/tmp/metricbeat-7.4.0-x86_64.rpm' do
        source 'https://client.com/ibm/chef-client/metricbeat-7.4.0-x86_64.rpm'
        owner 'root'
        group 'root'
        mode '0755'
        action :create_if_missing
    end

    # Install metricbeats rpm
	execute 'install metricbeats' do
        command 'rpm -ivh /tmp/metricbeat-7.4.0-x86_64.rpm'
        action :run
        only_if { shell_out("rpm -qa metricbeat").stdout == ''}
    end

    # Copy metricbeats config
    cookbook_file '/etc/metricbeat/metricbeat.yml' do
      source 'metricbeat.yml'
      mode '0600'
      action :create
    end

    # Copy metricbeats config
    cookbook_file '/etc/metricbeat/modules.d/system.yml' do
      source 'system.yml'
      mode '0600'
      action :create
    end

    service 'metricbeat' do
      supports :status => true
      action :start
    end
end